code-challenges.test.js

```diff
@@ -34,21 +110,130 @@ const object3 = { number: -7 }
+
+// Input 2:  randomNouns2
+// Output 2: ["Temperature", "Database", "Chopsticks", "Mango"]
+
+// Create a function called capitalizedArray that takes in an array of words and returns an array with all the words capitalized.
+// One parameter: array
+// Create a for loop
+    // for every string, capitalize the first letter and push it into a newArray
+        // return newArray
+
+describe("capitalizedArray", () => {
+    it("takes in an array of words and returns an array with all the words capitalized.", () => {
+      expect(capitalizedArray(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+      expect(capitalizedArray(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+    })
+  })
+
+// Got a good fail! Yay!
+
 const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
 const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
 
 // b) Create the function that makes the test pass.
 
+// Pseudocode:
+
+// I am going to write the template of a function so that I know what needs to be filled in based on the instructions I gave myself on lines 153-157.
+// I called my function capitalizedArray just as I instructed to do so on line 153.
+// Once I completed filling out the first line of my function, I needed to begin the logic portion of the function.
+// I needed to create a newArray which will house the outcome of my function.
+// I knew I needed to write out a for loop because I needed to iterate the array.
+  // I the variable word be the element of the index in the array. I established this because it makes it easier to read when I am coding rest of the for loop.
+// I knew that the built in method, .UpperCase(), capitalized letters of a string within an array. But I did not know how to access only the first letter.
+  // I googled how to capitalize the first letter in a string in Javascript. And learned about the built in method, .charAt().
+// .charAt() seperates a letter at the position given. For example: "Bear" --> .charAt(1) // Outcome: "e"
+// .slice() seperates the strings into two at a given index. For exmaple: "Bear" ---> .slice(1) // Outcome: "ear"
+// If I combine these two built in methods along with .UpperCase() I can seperate the letters so that I only capitalize the first one.
+// newArray.push is moving the outcome to newArray with capitalized letters.
+// Then I return the newArray
+
+
+const capitalizedArray = (array) => {
+    newArray = []
+    for(i=0; i<array.length; i++) {
+        let word = array[i]
+        newArray.push(word.charAt(0).toUpperCase() + word.slice(1))
+    } 
+    return newArray
+}
+
+console.log(capitalizedArray(randomNouns1))
+console.log(capitalizedArray(randomNouns2))
+
+// Passed the test! Yay!
```

Comment:
Awesome work!  If you are looking to stretch yourself, I would encourage you to try using a higher-order function when possible.